src/lib/containers/entity_finder/tree/TreeNode.tsx

```diff
@@ -44,17 +45,23 @@ export const TreeNode: React.FunctionComponent<TreeNodeProps> = ({
   autoExpand = () => false,
   visibleTypes = [EntityType.PROJECT, EntityType.FOLDER],
   rootNodeConfiguration,
-  appearance,
+  treeNodeType,
   selectableTypes,
+  currentContainer,
 }: TreeNodeProps) => {
   const isRootNode = !!rootNodeConfiguration
+  const nodeId = isRootNode ? 'root' : entityHeader!.id
+
+  const isSelected =
+    treeNodeType === EntityTreeNodeType.SELECT
+      ? selected.has(nodeId)
+      : currentContainer === nodeId
```

Comment:
If it's SELECT/single pane mode, check `selected`, if it's BROWSE/dual-pane mode, check `currentContainer`